Extension/ui/settings.ts

```diff
@@ -0,0 +1,104 @@
+        document.getElementById(elementId.compilerPath).addEventListener("change", this.onChanged.bind(this, elementId.compilerPath));
+        document.getElementById(elementId.intelliSenseMode).addEventListener("change", this.onChanged.bind(this, elementId.intelliSenseMode));
+
+        document.getElementById(elementId.includePath).addEventListener("change", this.onChanged.bind(this, elementId.includePath));
+        document.getElementById(elementId.defines).addEventListener("change", this.onChanged.bind(this, elementId.defines));
+
+        document.getElementById(elementId.cStandard).addEventListener("change", this.onChanged.bind(this, elementId.cStandard));
+        document.getElementById(elementId.cppStandard).addEventListener("change", this.onChanged.bind(this, elementId.cppStandard));
+
+        // document.getElementById(ElementId.compilerPathInvalid).style.visibility = "hidden";
+        // document.getElementById(ElementId.includePathInvalid).style.visibility = "hidden";
+    }
+
+    private onChanged(id: string) {
+        if (this.updating) {
+            return; 
+        }
+
+        var x = <HTMLInputElement>document.getElementById(id);
+        this.vsCodeApi.postMessage({
+            command: "change",
+            key: id,
+            value: x.value
+        });
+    }
+
+    private onMessageReceived(e: MessageEvent) {
+        const message = e.data; // The json data that the extension sent
+        switch (message.command) {
+            case 'update':
+                this.update(message.config);
+                break;
+            //TODO: validate input paths
+            // case 'validateCompilerPath':
+            //     this.validateInput(ElementId.compilerPathInvalid, message.invalid);
+            //     break;
+            // case 'validateIncludePath':
+            //     this.validateInput(ElementId.includePathInvalid, message.invalid);
+            //     break;
+        }
+    }
+
+    private update(config: any) {
+        this.updating = true;
+        try {
+            document.getElementById(elementId.activeConfig).innerHTML = config.name;
+
+            (<HTMLInputElement>document.getElementById(elementId.compilerPath)).value = config.compilerPath;
+            (<HTMLInputElement>document.getElementById(elementId.intelliSenseMode)).value = config.intelliSenseMode;
+
+            document.getElementById(elementId.includePath).innerHTML = (config.includePath.length > 0) ? config.includePath.join("\n") : "";
+            document.getElementById(elementId.defines).innerHTML = (config.defines.length > 0 ) ? config.defines.join("\n") : "";
+
+            (<HTMLInputElement>document.getElementById(elementId.cStandard)).value = config.cStandard;
+            (<HTMLInputElement>document.getElementById(elementId.cppStandard)).value = config.cppStandard;
+        }
+        finally {
+            this.updating = false;
+        }
+    }
+
+    // private validateInput(elementID: string, invalid: boolean) {

```

Comment:
Unused?

Reply:
Yes, keeping some of the code to validate inputs as I'm planning to work on them after insiders.